clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -2370,22 +2013,17 @@ Expected<SmallVector<StringRef>> linkAndWrapDeviceFiles(
         return FileNameOrErr.takeError();
       InputFiles.emplace_back(*FileNameOrErr);
     }
+
+    // Link the remaining device files using the device linker.
+    Expected<StringRef> OutputOrErr =
+        HasSYCLOffloadKind ? sycl::linkDevice(InputFiles, LinkerArgs)
```

Comment:
That's is wrong. We need to call both of these functions depending on the `HasSYCLOffloadKind` and `HasNoSYCLOffloadKind`. These two variables are independent of each other!

Reply:
Ah..sorry. I missed this on my 1-1 with Alexey Sachkov. Yes. There is a possibility that both HasSYCLOffloadKind and HasNonSYCLOffloadKind are true. So, there is a possibility that both these functions need to run

Reply:
And we are also missing linkBitcodeFiles call here. Are we not?